eclass/distutils-r1.eclass

```diff
@@ -1985,12 +1985,35 @@ _distutils-r1_post_python_install() {
 			examples test tests
 			.pytest_cache .hypothesis _trial_temp
 		)
+		local modname=$(get_modname)
+		local strays=()
 		local p
 		for p in "${forbidden_package_names[@]}"; do
-			if [[ -d ${sitedir}/${p} ]]; then
-				die "Package installs '${p}' package which is forbidden and likely a bug in the build system."
-			fi
+			[[ -d ${sitedir}/${p} ]] && strays+=( "${p}" )
 		done
+		while read -d '' -r p; do
+			case ${p} in
+				*.egg-info|*.pth|*.py|*.pyi|*${modname})
+					;;
+				*)
+					strays+=( "${p##*/}" )
+					;;
+			esac
+		done < <(find "${sitedir}" -maxdepth 1 -type f -print0)
```

Comment:
Can't we pass iname or something instead and then `readarray` the result, so we can skip the while loop?

Reply:
We could but it's easier and more readable this way. I really hate covering all of thse with `-o -name -o -name`.

Reply:
Fine with me then, thanks

Reply:
Actually, maybe you're right. We already have one big `find` like that, so I guess I should go for consistency.